Combine variables that provide user information

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request is mainly about removing the `non_self_admin_users` variable and expanding the `users` variable to include the same functionality. It also updates the `type` constraint for the `users` variable to reference the supported attributes. Lastly it updates the tfvars example in the README to mirror what is given in the example for the `users` variable.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

This idea was mentioned by @jsf9k in https://github.com/cisagov/cyhy-users-non-admin/pull/5#discussion_r819646945 and was met with team agreement. Thus I am implementing it in this pull request.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass. I updated the Production environment tfvars to reflect the changes in this pull request and a `terraform plan` shows no changes for existing users.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

mcdonnnj committed:
Consolidate variables that provider users

Remove the `non_self_admin_users` variable and integrate the same
functionality into the `users` variable by adding a new `self_managed`
attribute.

mcdonnnj committed:
Update the tfvars example in the README

Update the tfvars example that is in the README to mirror the example
given in the `users` variable description.

mcdonnnj committed:
Be more specific with the `users` variable type

Use an `map(object())` type with the specific attributes that are
supported instead of a blanket `map(map())` type.

jsf9k reviewed (approved):
Nice work!

dav3r reviewed (changes requested):
This is great!

I would like to see one small change to a pair of comments - add something like `self_managed is true and...` before the `require_mfa` clauses below:

https://github.com/cisagov/cool-users-non-admin/blob/c4013621cd3e15e5ffdf6ccf9f7f5da3702f5c24/users.tf#L10-L11

https://github.com/cisagov/cool-users-non-admin/blob/c4013621cd3e15e5ffdf6ccf9f7f5da3702f5c24/users.tf#L30-L31

mcdonnnj committed:
Update comments explaining policy attachments

Update the comments for the self_managed policies to reference the new
`self_managed` attribute in the `users` variable that is being used to
filter users.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r reviewed (approved):
🍰
